plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -584,24 +616,61 @@ public void setNull(PreparedStatement statement, int index)
         };
     }
 
-    public static LongWriteFunction trinoTimestampToOracleTimestampWriteFunction()
+    private static ObjectWriteFunction oracleLongTimestampWriteFunction(TimestampType timestampType)
+    {
+        int roundToPrecision = timestampType.getPrecision();
+        checkArgument(
+                MAX_SHORT_PRECISION <= roundToPrecision && roundToPrecision <= MAX_ORACLE_TIMESTAMP_PRECISION,
+                "Invalid roundToPrecision for %s: %s",
```

Comment:
How about `"Precision is out of range: %s", timestampType.getPrecision()`?

Also, I see that the same verification we are doing in `oracleLongTimestampReadFunction` and  `oracleLongTimestampWriteFunction`, We can extract it as a method and pass `timestampType` to verify it. Something like `verifyLongTimestampPrecision` WDYT?